scanner/decisions/0009-nvd-cvss-scores.md

```diff
@@ -0,0 +1,145 @@
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+      string updater = 3; <-- New field.
+      string cvss_url = 4; <-- New field
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+    repeated CVSS cvss_metrics = 13; <-- New field.
+  }
+  ...
+}
+```
+
+There will be a new type plus two new fields added:
+
+* `CVSS.updater`
+  * This specifies the source of the particular CVSS metrics. The value will be the name of the scanner updater, indicating the data source from which the updater is fetching vulnerabilities.
+* `CVSS.cvss_url`
+  * This specifies the CVSS score source URL, aiding both API and UI users in tracking the origin of this metric.
+* `cvss_metrics`
+  * This is a list of each unique CVSS metric based on the source.
+
+The original `cvss` field will remain and will continue to represent the Scanner's preferred CVSS score.
+This is currently the score from the vulnerability's original data source, if available, otherwise NVD.
+
+### Handling RHSA/RHBA/RHEA
+
+Without loss of generality, RHSA/RHBA/RHEA will just be referred to as the more well-known RHSA variant of the three.
+
+Scanner V4 currently shows RHSA as the top-level entity, rather than the related CVE(s), when the CVE(s) is/are fixed.
+When this is done, Scanner V4 gives the RHSA the highest CVSS score from the associated CVE(s). We acknowledge this is not
+ideal, and there are plans to resolve this in the future. For now, we will need to support NVD scores in a compatible manner.
+
+The [`claircore.Vulnerability.Severity`](https://github.com/quay/claircore/blob/v1.5.25/vulnerability.go#L24) is currently set to the following:
+
+`severity=<severity>&cvss3_score=<score3>&cvss3_vector=<vector3>cvss2_score=<score2>&cvss2_vector=<vector2>`
+
+This URL encoding will be extended to include `cve=<CVE ID>`.
+
+## Consequences
+
+* Introducing `string` field `CVSS.updater` ensures consistency and limits mistakes which may be made
+with misspelled or differently spelled strings because `updater` values directly come from Scanner vulnerability updater names.
+* `repeated CVSS` field `cvss_metrics` will always include CVSS metrics from all updaters/data sources, including the Scanner's preferred CVSS metric.
+This approach simplifies data querying and filtering, as `cvss_metrics` will be the sole field used for filtering data or making policies.
+* Encoding the RHSA/RHEA/RHBA's related CVE allows Scanner V4 to relate the advisory back to the CVE which has the highest score and search NVD for that CVE's score.
```

Comment:
Might make more sense in the RHSA section above, unless it's already mentioned there

Reply:
Move to the above section.